bin/create_dataset.py

```diff
@@ -0,0 +1,193 @@
+    """
+    Parse the xml file and return a dataframe with the articles and titles.
+
+    :param file_location: The location of the xml file
+    :returns: A dataframe with the articles and titles
+    """
+
+    logger.info(f"Reading file {file_location}")
+
+    tree = ET.parse(file_location)
+    root = tree.getroot()
+
+    titles_list = []
+    article_list = []
+
+    for i, page in enumerate(
+        root.findall("{http://www.mediawiki.org/xml/export-0.10/}page")
+    ):
+        title = ""
+        for p in page:
+            if p.tag == "{http://www.mediawiki.org/xml/export-0.10/}title":
+                title = p.text
+            if p.tag == "{http://www.mediawiki.org/xml/export-0.10/}revision":
+                for x in p:
+                    if x.tag == "{http://www.mediawiki.org/xml/export-0.10/}text":
+                        article_txt = x.text
+
+        if article_txt and title:
+            article_list.append(article_txt)
+            titles_list.append(title)
+
+    logger.info(
+        f"Successfully finished reading file. Found {len(article_list)} articles."
+    )
+
+    return pd.DataFrame((article_list, titles_list)).T.rename(
+        columns={0: "article", 1: "title"}
+    )
+
+
+def find_categories(df: pd.DataFrame, parent_category: str) -> pd.Series:
+    """
+    Find the categories for the parent category.
+
+    :param df: The dataframe with the articles and titles
+    :param parent_category: The parent category to find the categories for
+    :returns: A series with the categories for the parent category
+    """
+
+    logger.info(f"Finding categories for {parent_category}")
+    rows = df[df["article"].str.contains(f"Category:{parent_category}")]
+    result = rows["title"]
+
+    logger.info(f"Found {len(result)} categories for {parent_category}")
+
+    return result
+
+
+def find_articles(df: pd.DataFrame, category: str):
+    """
+    Find the articles for the category.
+
+    :param df: The dataframe with the articles and titles
+    :param category: The category to find the articles for
+    :returns: A series with the articles for the category
+    """
+
+    logger.info(f"Finding articles for {category}")
+
+    result = df["article"].str.contains(f"{category}")
+
+    logger.info(f"Found {result.sum()} articles for {category}")
+
+    return result
+
+
+def main(input_path: str, output_path: str, parent_category: str):
+    raw_df = parse(input_path)
+
+    df = raw_df.copy()
+
+    categories = find_categories(raw_df, parent_category)
+
+    df["category"] = pd.NA
+
+    for category in categories:
+        articles = find_articles(raw_df, category)
+        df.loc[articles, "category"] = category
+
+    df = df.dropna(subset=["category"])
+
+    logger.info(f"Found {len(df)} articles for {parent_category} category.")
+
+    df = clean_df(df)
+
+    logger.info(f"Writing to file {output_path}")
+    df.to_json(output_path, orient="records", lines=False)
```

Comment:
Make json properly formatted here, so that its human readable (probably some option like lines=True or indet=2/4) - idk but one liner fix from documentation